src/peft/utils/save_and_load.py

```diff
@@ -335,10 +335,11 @@ def set_peft_model_state_dict(
     state_dict = {}
     if getattr(model, "modules_to_save", None) is not None:
         for key, value in peft_model_state_dict.items():
-            if any(module_name in key for module_name in model.modules_to_save):
-                for module_name in model.modules_to_save:
-                    if module_name in key:
-                        key = key.replace(module_name, f"{module_name}.modules_to_save.{adapter_name}")
+            if any(f".{module_name}." in key for module_name in model.modules_to_save):
```

Comment:
This does not cover the case where the module uses a `nn.Parameter` instead of `nn.Linear` as classification head. Do we want to support that as well?

Reply:
The targeted object for `modules_to_save` must be an `nn.Module`, it's not possible to target an `nn.Parameter`. Or am I misunderstanding?

Reply:
Ah, perfect, nevermind then.